frontend/src/utils/address.ts

```diff
@@ -5,13 +5,14 @@
 import { CnsQueryResponse, Provider } from 'components/models';
 import { utils } from '@umbra/umbra-js';
 import { MAINNET_PROVIDER } from 'src/utils/constants';
-import { getAddress, Web3Provider } from 'src/utils/ethers';
+import { getAddress, Web3Provider, isHexString } from 'src/utils/ethers';
 import { getChainById } from 'src/utils/utils';
 import { i18n } from '../boot/i18n';
 // ================================================== Address Helpers ==================================================
 
 // Returns an address with the following format: 0x1234...abcd
-export const formatAddress = (address: string) => `${address.slice(0, 6)}...${address.slice(38)}`;
+export const formatNameOrAddress = (address: string) =>
+  isHexString(address) ? `${address.slice(0, 6)}...${address.slice(38)}` : address;
```

Comment:
Nit on naming/style:

```suggestion
export const formatNameOrAddress = (nameOrAddress: string) => {
  return isHexString(nameOrAddress) ? `${nameOrAddress.slice(0, 6)}...${nameOrAddress.slice(38)}` : nameOrAddress;
}
```